rust/storage/src/lib.rs

```diff
@@ -62,6 +61,24 @@ impl ChromaError for PutError {
 }
 
 impl Storage {
+    pub fn new_test_storage() -> Self {
+        Storage::Local(LocalStorage::new(
+            TempDir::new()
+                .expect("Should be able to create a temporary directory.")
+                .into_path()
+                .to_str()
+                .expect("Should be able to convert temporary directory path to string"),
+        ))
+    }
+
+    pub fn new_test_storage_at<P: AsRef<Path>>(path: P) -> Self {
```

Comment:
Can I recommend making a determinstic tempdir that is formed by doing `file!()` and `line!()`?  That way we know exactly which temporary directory corresponds to which test.  I'd use a macro_rules so that it picks up the caller's file/line.

Reply:
not opposed to this, but wondering how this would be useful?

Reply:
Test fails and there's a deterministic object storage location to look at the files.  Otherwise you have to print the path in this function, look at the log output, etc.  Seeing the name of the file or function name (or anything marking the function) does the trick.

Here's where I did this before:  https://github.com/rescrv/blue/blob/main/mani/src/lib.rs#L721

Reply:
for that to be usable, we would either have to print out the temp path or create the temp path under the working directory... I'm still somewhat struggling to see how this would be useful since block objects are not really analyzable by themselves without additional tooling

Reply:
We don't have the additional tooling.  But when we need it, we'll have this proposed change plus that tooling to write.  Just one future annoyance we might be able to cut out.

This is a nit.  Don't hang the merge on this feedback, but consider it if it's easy.